### PR TITLE
[feat] : Improved Register.jsx file

### PR DESCRIPTION
**Pull request by Yash-Vashishth**
Hi @dhairyagothi !! This pull request resolves the issue listed in issue #225. 

I have added FormData component, changed formState components to allow better organization of the code overall and maintainability and added some password strength feedbacks andloading indicators. **Please** check the changes made in this file and ***UPGRADE THE LEVEL OF THE ISSUE*** if possible. Also commit it to the main branch if you find the changes satisfactory.

## Type of Pull Request

- [ ] Bug Fix
- [X] New Feature
- [X] Performance Improvement
- [ ] Documentation Update
- [ ] Code Refactoring
- [ ] Security Enhancement
- [ ] Other (Specify): _______________

(Please check relevant boxes by entering an "X" inside them)

## Checklist for Contributor
use'X' in [ ] to check it .

- [X] I have reviewed my code before submission.
- [X] I have read and followed the project's Contribution Guidelines.
- [X] I have tested the code thoroughly.
- [X] I have attached related issue numbers, screenshots, or videos (if applicable).
- [X] I have added comments to my code where necessary for clarity.
- [X] I have checked the existing code base and my changes do not introduce any breaking changes.
- [X] My PR addresses only one issue; I have created separate PRs for other issues.
- [X] I have ensured that the PR does not include unrelated files.
- [X] I am mindful of the project's coding standards and best practices.

Thanks,
Yash Vashishth
(Contributor, GSSoC'24 and Hacktoberfest)

### CODE REVIEW

**Comment by dhairyagothi** [comment]
image

@Yash-Vashishth please check there is some problem in routes

**Comment by Yash-Vashishth** [comment]
Hi @dhairyagothi !! I just checked the commit I have made and wanted to clarify that the routes for register.jsx have not been changed in any way and are the same. I have just changed the code structure, physical elements and presentation, the internal information has not been touched in any way. Please check

File: frontend/src/Pages/Register.jsx [hunk 1]
```diff
@@ -6,62 +6,73 @@ import { GoogleLogin } from '@react-oauth/google';
...
+            placeholder={placeholder}
+            pattern={pattern}
+            maxLength={maxLength}
+            className="w-full px-3 py-2 text-sm border border-gray-300 rounded-md focus:outline-none focus:ring-2 focus:ring-blue-500"
+            required={required}
+        />
+        {errorMessage && <p className="text-sm text-red-500">{errorMessage}</p>}
+    </div>
+);
+
```
Comment (dhairyagothi) [inline frontend/src/Pages/Register.jsx]: i think because of this function there is problem

**Comment by dhairyagothi** [comment]
@Yash-Vashishth conflicts

**Comment by Yash-Vashishth** [comment]
Will resolve by today

**Comment by Yash-Vashishth** [comment]
@dhairyagothi pls check the changes now and see if the code is working fine or not

**Review by dhairyagothi — APPROVED** [review]
good to go now

**Comment by Yash-Vashishth** [comment]
Bro level2 tag pls

**Pull request merged**